### PR TITLE
Backend STI, policy, search, controller, validations etc for EarlyAccess

### PR DESCRIPTION
**Pull request by bkmaibach**
## Description

Backend work for re-using RequirementTemplate model to handle the concept of previews.
Adds new fields for blocks and requirement templates
Adds early_access to frontend search model and search concern for obtaining records considered early access.

This builds on the sandbox backend work, but disables it on the front end as this feature is delayed until after super admin early access feature is complete.

### CODE REVIEW

File: app/frontend/components/domains/super-admin/early-access/requirements-library/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            <Text color={"text.secondary"} mt={1}>
+              {t("earlyAccessRequirementsLibrary.index.description")}
+            </Text>
+          </Box>
+          {/* <RequirementsBlockModal /> */}
+        </Flex>
+
+        {/* <RequirementBlocksTable alignItems={"flex-start"} w={"full"} /> */}
+
+        {/* <ToggleArchivedButton searchModel={requirementBlockStore} mt={3} /> */}
```
Comment (LaurelOlson) [inline app/frontend/components/domains/super-admin/early-access/requirements-library/index.tsx:28]: crud? or TODO?

File: app/models/early_access_requirement_template.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class EarlyAccessRequirementTemplate < RequirementTemplate
+  # has_one :template_version, dependent: :destroy, foreign_key: :requirement_template_id
```
Comment (LaurelOlson) [inline app/models/early_access_requirement_template.rb:2]: crud?

File: app/frontend/components/domains/super-admin/early-access/requirements-library/index.tsx (hunk 1 above)
Comment (bkmaibach) [reply app/frontend/components/domains/super-admin/early-access/requirements-library/index.tsx:28]: todo - see later branches

File: app/models/early_access_requirement_template.rb (hunk 2 above)
Comment (bkmaibach) [reply app/models/early_access_requirement_template.rb]: TODO - fixed later

**Review by LaurelOlson — APPROVED** [review]

**Pull request merged**